### ISSUE TITLE
BUG: listener registration not works due Select2

### ISSUE THREAD

**Issue by JiriCepelkaFirstLineSoftware** [select2/select2#5420]
I found Select2 preventing registered event listener to be called. There is [duplicate issue](https://github.com/select2/select2/issues/5390) to this but without any details or code samples.

Issue – onchange or addEventListener listener on form that employs Select2 is never called.

### **Example:**

In example you can observe this:

1. There are six forms.
2.  Three of them employs Select2 (first three).
3. Only one (first) form of first three ones with inline registration calls the listener.
4. Rest of the forms are the same as the first three except they don’t employ Select2. They serves verification purpose.

```html
<html>

<head>
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <link href="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.5/css/select2.min.css" rel="stylesheet" />
  <script src="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.5/js/select2.min.js"></script>
  <script>

        $(document).ready(function() {
          $('.stoppingPropagationElem').select2();
        });
    
        function onChange_Function(obj) {
    
          let text;
          if (obj.currentTarget != undefined) {
            text = obj.currentTarget.name;
          } else {
            text = obj;
          }
    
          document.body.appendChild(getPwithText(text));
        }
    
        function getPwithText(text) {
          let node = document.createElement("P");
          let textnode = document.createTextNode(text);
    
          node.appendChild(textnode);
    
          return node;
        }
    
        window.addEventListener("load", function() {
          document.forms["someForm2"].onchange = onChange_Function;
          document.forms["someForm3"].addEventListener("change", onChange_Function);
    
          document.forms["someForm5"].onchange = onChange_Function;
          document.forms["someForm6"].addEventListener("change", onChange_Function);
        });
  </script>
</head>

<body>
  <form name="someForm" onchange="onChange_Function(this.name)">
    <select class="stoppingPropagationElem">
      <option>1</option>
      <option>2</option>
      <option>3</option>
    </select>
  </form>
  <form name="someForm2">
    <select class="stoppingPropagationElem">
      <option>1</option>
      <option>2</option>
      <option>3</option>
    </select>
  </form>
  <form name="someForm3">
    <select class="stoppingPropagationElem">
      <option>1</option>
      <option>2</option>
      <option>3</option>
    </select>
  </form>
  <form name="someForm4" onchange="onChange_Function(this.name)">
    <select>
      <option>1</option>
      <option>2</option>
      <option>3</option>
    </select>
  </form>
  <form name="someForm5">
    <select>
      <option>1</option>
      <option>2</option>
      <option>3</option>
    </select>
  </form>
  <form name="someForm6">
    <select>
      <option>1</option>
      <option>2</option>
      <option>3</option>
    </select>
  </form>
</body>

</html>
```

StackOverflow: https://stackoverflow.com/questions/53304417/js-onchange-works-addeventlistener-does-not

JSFiddle: https://jsfiddle.net/azgsvuj5/.

**Comment by JiriCepelkaFirstLineSoftware** [comment]
No reaction and self closing? That not what I expected.

**Comment by os1r1s110** [comment]
It's been marked as stale by a bot, not by the repo's owner, just by adding a new comment, you made it stay alive for another while.

**Comment by kevin-brown** [comment]
This one is... odd. Does anyone have a guess as to why it happens?

For me, when testing, the first one (where the event handler is bound to the `<form>`) works but the other two (where the event handler is bound to the `<select>`) do not.

**Comment by orthener** [comment]
https://makandracards.com/makandra/71890-events-triggered-by-jquery-cannot-be-observed-by-native-event-listeners
Changed `select2.js` file in project:
replaced
```
this.$element.trigger('change');
````
to
```
let eventChange = new CustomEvent('change');
self.$element[0].dispatchEvent(eventChange);
```
And all works well

**Comment by asadakbar** [comment]
Why close this if it is a confirmed bug? Maybe stalebot needs to turn itself off if the "confirmed bug" label is applied so at least its clear to folks this is a continuing issue.

FWIW, I used the example in the makandra link provided by @orthener and updated it to use native dispatch and a few other tweaks.

````
jQuery.fn.enableNativeEvent = function(eventNames, convertedPropNames = []) {
  eventNames.forEach(eventName => {
    this.on(eventName, function($event) {
      if (!$event.originalEvent) {
        $event.preventDefault();
        const newName = `$:${eventName}`;
        const eventData = _.pick($event, convertedPropNames) // Uses lodash.pick
        const event = new CustomEvent(newName, 
          { detail: eventData, bubbles: true, cancelable: true  }
        )
        $event.target.dispatchEvent(event)
      }
    })
  })
  return this
}

// Usage
$(".select2able").select2().enableNativeEvent(["change.select2"], ["params"]);
````
